### ISSUE TITLE
Active folder isn't set properly when navigating with the mouse in Column View

### ISSUE THREAD

**Issue by alahmnat** [files-community/Files#15869]
### Description

In Column View, clicking on a folder doesn't move the target for interactions into that folder, even though the tab name is updated to reflect the selected folder. The interaction target remains in the column where that folder is listed. As an example, in the image below I've clicked on the PowerShell folder and the tab name says "PowerShell", but the path bar still indicates I'm in the Documents folder, and if I paste a file at this point, it'll get pasted into Documents, not PowerShell. 

![image](https://github.com/user-attachments/assets/1070a042-04bf-459a-bb20-f1ffbf34e103)

Clicking into the folder's column (in the example above, clicking anywhere in the "Scripts" column) is required to actually update the active folder. Essentially, the "active" folder is always the column you've most recently clicked on. This also applies for going "up" the hierarchy to the left; clicking on a folder inside of "Scripts" will advance the target to Scripts, but clicking on "PowerShell" two columns back will move the target all the way back to Documents, not PowerShell.

When navigating with the arrow keys, this makes some degree of sense as the focus ring doesn't actually move into the selected folder until you hit the left or right arrow key (though in that case the tab name is updating incorrectly because the selected folder isn't your current "active" location yet):

![image](https://github.com/user-attachments/assets/11dddf39-3931-4854-981f-31513ad24ee2)

### Mouse Interaction
1. Be in Column View.
2. Click on a folder within a column - selection depth doesn't seem to matter.
3. Observe tab title changes to match the name of the selected folder.
4. Paste a file - it goes into the column with the folder you clicked on.
5. Click on anything in the column showing step 2's folder's contents - whitespace, files, or even other folders.
6. Paste a file - it goes into the folder selected in step 2.
7. Click on an ancestor folder.
8. Observe the path and interaction target become the column with the selected folder, not the selected folder itself.

### Keyboard Interaction
1. Be in Column View.
2. Use arrow keys to navigate vertically through a column's content.
3. Observe tab title changes to match the name of the highlighted folder.
4. Paste a file - it goes into the column with the highlighted folder (the currently "active" column), not the highlighted folder itself.
5. Press right arrow to navigate into the folder with the focus ring.
6. Paste a file - it goes into the column you've moved your selection into.

### Requirements

- Clicking on a folder in Column View should update all of the following to match the folder that's been clicked on, whether moving to the right or the left along the column hierarchy:
  1. Tab title
  2. Path bar location
  3. Interaction target (pasting, etc.)
- Using arrow keys to navigate in Column View should do either of the following:
  1. Follow mouse behavior, so tab title, path bar location, and interaction target advance to the highlighted folder's column but arrow navigation itself is unaffected (this is how Finder works in macOS, incidentally), **or**
  2. Tab title should reflect the folder name of the current column (where the focus ring is) and not update until using the right or left arrow key to move into or out of a folder.

In the case of a column that is floor-to-ceiling full of folders (so there's no files or whitespace to click on), one would need to click on the parent folder to be able to paste into it. (macOS tries to mitigate this particular case by not having the click target extend across the entire column, so you can click on the whitespace next to a folder's name to select that column, which is itself quirky.) Having to technically click on a parent to paste into it may feel slightly unintuitive at first blush, but IMO it's less unintuitive than "clicking on a folder selects its parent", as is currently the case. 

I guess what I'm really asking is for mouse behavior to be more intuitive (clicking on a folder selects *it*, not its column) rather than following the current keyboard navigation logic, and for the keyboard navigation logic to match this new mouse behavior to be consistent.

### Files Version

3.5.0.0

### Windows Version

10.0.22631.3880

### Log File

[debug.log](https://github.com/user-attachments/files/16382218/debug.log)

**Comment by yaira2** [comment]
Is the expected behavior for the focus to switch to the newly opened folder when opened via mouse input?

**Comment by yaira2** [comment]
If anyone has Finder on Mac, it would be helpful to know how they handle this.

**Comment by alahmnat** [comment]
Yeah, the tl;dr is to switch to the folder you click on when you click on it, which is how Finder operates. The tab title in Files changes to indicate the selected folder, but not the rest of the app (which just adds to the confusion). I've pasted stuff into the parent folder of where I'm trying to put things more times than I can count ;)

**Comment by yaira2** [comment]
Thanks!

**Comment by hishitetsu** [comment]
I checked the behavior on a Mac and found that when I selected a folder with the mouse, the column that opened that folder didn't get focus, but the column of the parent folder remained in focus. But the paste operation is performed on the selected folder, not the parent folder.

**Comment by yair100** [comment]
@hishitetsu which behavior do you prefer?

**Comment by hishitetsu** [comment]
I think it should behave the same as on a Mac. Couldn't we apply `PasteItemToSelectionAction` instead of `PasteItemAction` in the columns view?

**Comment by yair100** [comment]
Pasting content is just one example, it's still an issue for other actions. It's also confusing to treat the second to last column as the active one when the user opened a different folder.

**Comment by natedog2424** [comment]
I would like to express my desire to see this feature implemented as it feels quite counterintuitive to me at the moment. 
The biggest issue I see is with the Git integration. When clicking on a folder containing a git repo I can see the entire contents, but none of the git stuff shows up. I personally think that the address bar should be updated to show whatever folder you have clicked on last in columns view especially when considering the "Open folders with a single click" setting which tells us that in columns view the folder should be "opened" with a single click and opening a folder in any other view sets the address bar to point to that folder. If the click behavior was update to more closely reflect opening the folder like the setting says then I think that the rest of the things like pasting and the git integration would be resolved naturally as a byproduct. While I'm talking about the columns view it would be nice to see visually what parent folder you came from :)

**Comment by stanlrt** [comment]
The issue is not that the focus does not switch to the child folder (here "Script"), but that the pathbar does not show the path of the focused directory ("PowerShell").

cc @Josh65-2201

**Comment by yair100** [comment]
The path in the Omnibar is based on the focused directory. Fixing this should take care of #18563.
